Add recipe for arXiv-citation

slotThe opened the pull request:
### Brief summary of what the package does

Utility functions for dealing with arXiv papers.  It can generate citation data for PDF files from the arXiv and, optionally, download preprints to a specified directory and open them.  It also includes integration into `elfeed`. 

(If needed, a longer explanation, as well as preview GIFs, can be found in the README)

### Direct link to the package repository

https://gitlab.com/slotThe/arxiv-citation

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback

NOTE: There are two false positives

    157:14: error: `authors' was removed in Emacs version 25.1.
    218:10: error: `authors' was removed in Emacs version 25.1.

that I left in since I like the name of the variables in these cases.

- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thank you for this!  Very quick first pass --

It's unconventional for package names to have mixed case and there is a **strong** preference for all-downcased package names that you should adhere to unless there's a compelling reason.  E.g. "[latex](https://melpa.org/#/?q=latex)" instead of "LaTeX".

`arXiv-citation.el` with byte-compile using Emacs 27.1:
```
arXiv-citation.el:150:15:Warning: assignment to free variable
    `bibtex-autokey-year-title-separator'
arXiv-citation.el:151:15:Warning: assignment to free variable
    `bibtex-autokey-titleword-separator'
In arXiv-citation-elfeed:
arXiv-citation.el:293:56:Warning: reference to free variable
    `elfeed-show-entry'
In end of data:
arXiv-citation.el:297:1:Warning: the following functions are not known to be
    defined: bibtex-generate-autokey, elfeed-entry-link
```
I think these are solved by adding `(require 'bibtex)`, and you can pacify the byte-compiler against the `elfeed` warnings using `(defvar elfeed-show-entry)` and `defcustom` on elfeed-entry-link.


`arXiv-citation.el` with package-lint 20220412.1648:
```
2 issues found:
168:14: error: `authors' was removed in Emacs version 25.1.
229:10: error: `authors' was removed in Emacs version 25.1.
```
I read your note above.  A convention I picked up from Python is to just use e.g. `authors_` as the variable name instead, where the trailing `_` denotes "I don't want this to collide with some other thing with the same name".  I'll leave this up to you, but it can be nice to have a sparkling clean CI run so that you notice when actual issues arise.

Package and license:
- Add a GPL-compatible LICENSE file to the repository

slotThe commented:
@riscy Thank you for the review!

> It's unconventional for package names to have mixed case and there is a **strong** preference for all-downcased package names that you should adhere to unless there's a compelling reason. E.g. "[latex](https://melpa.org/#/?q=latex)" instead of "LaTeX".

Mh, I see.  I was more looking towards things like `TeX-mode` and `LaTeX-mode` when deciding upon a name, but I suppose these are the exception rather than the rule.  The arXiv _is_ written like that (well, really, the capital X is a capital greek "chi", but let's not get into that :) but I suppose so is LaTeX...  I don't really have any strong opinions here, so if there is a strong preference towards all-lowercase names then so be it—I've changed it now.

> `arXiv-citation.el` with byte-compile using Emacs 27.1:
>
> […]

Huh, curious.  Is there anything more to do than to run `byte-compile-file` on the file?  Because I definitely did that and it did not generate any of these warnings; sort of embarrassed now that I ticked that box without making sure—sorry about that :/  In either case, thank for double checking, should all be done now!

> `arXiv-citation.el` with package-lint 20220412.1648:
>
> […]
>
> I read your note above. A convention I picked up from Python is to just use e.g. `authors_` as the variable name instead, where the trailing `_` denotes "I don't want this to collide with some other thing with the same name". I'll leave this up to you, but it can be nice to have a sparkling clean CI run so that you notice when actual issues arise.

That sounds reasonable, fixed.

> Package and license:
>
>     * Add a GPL-compatible LICENSE file to the repository

Oh, I thought the note in the file would suffice; I added the full GPL3 as a LICENSE file to the repo as well.

riscy commented:
>  I don't really have any strong opinions here, so if there is a strong preference towards all-lowercase names then so be it—I've changed it now

There's probably no *real* harm but it appears to be the convention, and I'm a pedantic hobgoblin :)  Even `LaTeX-mode` is just an alias for the true function whose name is `latex-mode` (maybe you want to add an alias?)

> Is there anything more to do than to run byte-compile-file on the file?

This gets everyone, don't worry. :)  Most likely you're not seeing all of the compilation warnings because you're compiling your code in an instance of Emacs that has already loaded the code, or some past version of it at least.  Using `flycheck` helps with this, or you can restart emacs using `emacs -Q` and try compiling from there.

Looks good here now -- thanks, and welcome to MELPA!
